test/distributed/org/apache/cassandra/distributed/test/accord/AccordMetricsTest.java

```diff
@@ -0,0 +1,258 @@
+        }
+
+        assertCoordinatorMetrics(0, "ro", 0, 0, 0, 1, 0);
+        assertCoordinatorMetrics(1, "ro", 0, 0, 0, 0, 0);
+        assertReplicaMetrics(0, "ro", 0, 0, 0);
+        assertReplicaMetrics(1, "ro", 0, 0, 0);
+
+        assertZeroMetrics("rw");
+
+        countingMetrics0 = getMetrics();
+        try
+        {
+            SHARED_CLUSTER.coordinator(1).executeWithResult(writeCql(), ConsistencyLevel.ALL, 0, 0, 0, 0);
+        }
+        catch (RuntimeException t)
+        {
+            assertThat(t.getCause().getClass().getName()).isEqualTo(WriteTimeoutException.class.getName());
+        }
+
+        assertCoordinatorMetrics(0, "rw", 0, 0, 0, 1, 0);
+        assertCoordinatorMetrics(1, "rw", 0, 0, 0, 0, 0);
+        assertReplicaMetrics(0, "rw", 0, 0, 0);
+        assertReplicaMetrics(1, "rw", 0, 0, 0);
+
+        assertZeroMetrics("ro");
+    }
+
+    private void assertZeroMetrics(String scope)
+    {
+        for (int i = 0; i < SHARED_CLUSTER.size(); i++)
+        {
+            assertCoordinatorMetrics(i, scope, 0, 0, 0, 0, 0);
+            assertReplicaMetrics(i, scope, 0, 0, 0);
+        }
+    }
+
+    private void assertCoordinatorMetrics(int node, String scope, long fastPathTrx, long slowPathTrx, long preemptions, long timeouts, long recoveries)
+    {
+        DefaultNameFactory nameFactory = new DefaultNameFactory("accord-coordinator", scope);
+        Map<String, Long> metrics = diff(countingMetrics0).get(node);
+        logger.info("Metrics for node {} / {}: {}", node, scope, metrics);
+        Function<String, Long> metric = n -> metrics.get(nameFactory.createMetricName(n).getMetricName());
+        assertThat(metric.apply("FastPathTrx")).isEqualTo(fastPathTrx);
+        assertThat(metric.apply("SlowPathTrx")).isEqualTo(slowPathTrx);
+        assertThat(metric.apply("PreemptedTrx")).isEqualTo(preemptions);
+        assertThat(metric.apply("TimedoutTrx")).isEqualTo(timeouts);
+        assertThat(metric.apply("RecoveryDelay")).isEqualTo(recoveries);
+        assertThat(metric.apply("RecoveryTime")).isEqualTo(recoveries);
+        assertThat(metric.apply("Dependencies")).isEqualTo(fastPathTrx + slowPathTrx);
+    }
+
+    private void assertReplicaMetrics(int node, String scope, long commits, long executions, long applications)
+    {
+        DefaultNameFactory nameFactory = new DefaultNameFactory("accord-replica", scope);
+        Map<String, Long> metrics = diff(countingMetrics0).get(node);
+        Function<String, Long> metric = n -> metrics.get(nameFactory.createMetricName(n).getMetricName());
+        assertThat(metric.apply("CommitLatency")).isEqualTo(commits);
+        assertThat(metric.apply("ExecuteLatency")).isEqualTo(executions);
+        assertThat(metric.apply("ApplyLatency")).isEqualTo(applications);
+        assertThat(metric.apply("ApplyDuration")).isEqualTo(applications);
+        assertThat(metric.apply("Dependencies")).isEqualTo(executions);
```

Comment:
nit: For all of these metric names, we could make constants out of them in `AccordMetrics` to keep things DRY.

Reply:
Do you mean just those strings or full mbean names?

Reply:
Just the string literals here...but looks like you've already handled this.